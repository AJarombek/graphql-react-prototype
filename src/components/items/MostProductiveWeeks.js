/**
 * MostProductiveWeeks component which displays the weeks in the past year with the most
 * contributions (commits).
 * @author Andrew Jarombek
 * @since 4/21/2020
 */

import React, { useEffect, useState } from 'react';
import { getWeeklyContributionCounts } from '../../datasource/GraphQL';

const MostProductiveWeeks = () => {
  return (
    <div className="items most-productive-weeks">

    </div>
  );
};

export default MostProductiveWeeks;
