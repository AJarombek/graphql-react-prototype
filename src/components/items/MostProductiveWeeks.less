/**
 * Stylesheet for the MostProductiveWeeks component.
 * Author: Andrew Jarombek
 * Date: 4/21/2020
 */

@import "../../vars";

.most-productive-weeks {
  grid-area: prodWeeks;

  @media (max-width: 750px) {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  h2 {
    margin-bottom: 20px;
  }

  .week:nth-child(2) {
    p:nth-child(1) {
      font-size: 20px;
    }

    p:nth-child(2) {
      font-size: 20px;
      .hoverPurple();
    }
  }

  .week {
    display: flex;
    align-items: center;
    margin: 8px 0;
    width: 300px;

    p:nth-child(1) {
      font-size: 16px;
    }

    p:nth-child(2) {
      font-size: 18px;
      margin: 0 0 0 auto;
    }
  }
}
