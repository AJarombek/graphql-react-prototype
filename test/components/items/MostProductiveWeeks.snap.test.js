/**
 * Snapshot test for the MostProductiveWeeks component.
 * @author Andrew Jarombek
 * @since 4/25/2020
 */

import React from 'react';
import MostProductiveWeeks from '../../../src/components/items/MostProductiveWeeks';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<MostProductiveWeeks />).toJSON();
  expect(tree).toMatchSnapshot();
});
