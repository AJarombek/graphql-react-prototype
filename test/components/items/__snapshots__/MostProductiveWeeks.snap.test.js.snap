// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="items most-productive-weeks"
>
  <h2>
    Most Productive Weeks
  </h2>
</div>
`;
